test/devFunds.js

```diff
@@ -0,0 +1,139 @@
+const DevFunds = require('../lib/cmds/blockchain/dev_funds');
+const async = require('async');
+const FakeIpcProvider = require('./helpers/fakeIpcProvider');
+const utils = require('../lib/utils/utils');
+i18n.setOrDetectLocale('en');
+
+describe('embark.DevFunds', function () {
+  let config = {
+    networkType: 'livenet',
+    genesisBlock: 'foo/bar/genesis.json',
+    geth_bin: 'geth',
+    datadir: '/foo/datadir/',
+    mineWhenNeeded: true,
+    rpcHost: 'someserver',
+    rpcPort: 12345,
+    rpcApi: ['eth', 'web3', 'net', 'debug'],
+    rpcCorsDomain: true,
+    networkId: 1,
+    port: 123456,
+    nodiscover: true,
+    maxpeers: 25,
+    mine: true,
+    vmdebug: false,
+    whisper: false,
+    account: {
+      password: './test/test1/password',
+      numAccounts: 3,
+      balance: "5 ether"
+    },
+    bootnodes: "",
+    wsApi: ["eth", "web3", "net", "shh", "debug"],
+    wsHost: "localhost",
+    wsOrigins: false,
+    wsPort: 8546,
+    wsRPC: true,
+    targetGasLimit: false,
+    syncMode: undefined,
+    syncmode: undefined,
+    verbosity: undefined,
+    proxy: true
+  };
+
+  if (config.proxy) {
+    config.wsPort += constants.blockchain.servicePortOnProxy;
+    config.rpcPort += constants.blockchain.servicePortOnProxy;
+  }
+
+  describe('#create, fund, and unlock accounts', function () {
+    let provider = new FakeIpcProvider();
+    let devFunds = new DevFunds({blockchainConfig: config, provider: provider, logger: new TestLogger({})});
+    const web3 = new Web3(provider);
+
+    it('should create correct number of accounts', function (done) {
+      provider.injectResult(['0x47d33b27bb249a2dbab4c0612bf9caf4c1950855']); // getAccounts - return --dev account
+      devFunds.getCurrentAccounts(() => {
+
+        provider.injectResult('0x11f4d0a3c12e86b4b5f39b213f7e19d048276dae'); // createAccount #1
+        provider.injectResult('0x22f4d0a3c12e86b4b5f39b213f7e19d048276dab'); // createAccount #2
+
+        devFunds.createAccounts(config.account.numAccounts, 'test_password', (err) => {
+          assert.equal(err, null);
+
+          // TODO: make FakeIpcProvider smart enough to keep track of created accounts
+          provider.injectResult(['0x47d33b27bb249a2dbab4c0612bf9caf4c1950855', '0x11f4d0a3c12e86b4b5f39b213f7e19d048276dae', '0x22f4d0a3c12e86b4b5f39b213f7e19d048276dab']);
+
+          web3.eth.getAccounts().then((accts) => {
+            assert.equal(accts.length, config.account.numAccounts);
+            assert.strictEqual(accts[0], '0x47D33b27Bb249a2DBab4C0612BF9CaF4C1950855');
+            assert.strictEqual(accts[1], '0x11f4d0A3c12e86B4b5F39B213F7E19D048276DAe');
+            assert.strictEqual(accts[2], '0x22F4d0A3C12E86b4b5F39B213f7e19D048276DAb');
+            done();
+          });
+        });
+      });
+    });
+
+    it('should unlock accounts', function (done) {
+      provider.injectResult(true); // account #1 unlock result
+      provider.injectResult(true); // account #2 unlock result
```

Comment:
Just to make sure. If we run this test without the other one before, does it still work?

Reply:
This has been updated with `DevFunds` as sort of a factory pattern for construction, which allows for each unit test case to be run independent of the others. For clarity, there was not (and still is not) an issue with running unit test cases individually as a limitation of the `FakeIpcProvider`. The `FakeIpcProvider` just pushes results via `injectResults` and then when a call is made on the `web3` object, the `FakeIpcProvider` simply pops the latest result and returns it.